### PR TITLE
Add example compilation tracer to docs

### PR DESCRIPTION
**Pull request by wojtekmach**

### CODE REVIEW

File: lib/elixir/lib/code.ex [hunk 1]
```diff
@@ -85,6 +85,19 @@ defmodule Code do
 
   New events may be added at any time in the future, therefore it is advised
   for the `trace/2` function to have a "catch-all" clause.
+
+  Below is example tracer that prints all remote function invocations:
+
+      defmodule MyTracer do
+        def trace({:remote_function, _meta, module, name, arity}, env) do
+          IO.puts "#{env.file}:#{env.line} #{inspect(module)}.#{name}/#{arity}"
```
Comment (wojtekmach) [inline lib/elixir/lib/code.ex]: When testing this out on a sample project, I noticed `meta` contained `:line` but not `:file`, hence I ended up using macro env to grab both. Is this the way to go? Are there cases where `meta` contains data that is not in the env?

Comment (josevalim) [reply lib/elixir/lib/code.ex]: > Are there cases where meta contains data that is not in the env?

Yes, there is documentation about available metadata in the `Macro` module. The metadata may include more parsers information too, if you enable them in the compiler.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

File: lib/elixir/lib/code.ex (hunk 1 above)
Comment (wojtekmach) [reply lib/elixir/lib/code.ex]: Got it, thank you!

**Pull request merged**